src/osd/modules/render/bgfx/shaders/chains/crt-geom/fs_crt-geom-deluxe.sc

```diff
@@ -140,16 +146,67 @@ vec4 scanlineWeights(float distance, vec4 color)
   // "weights" should have a higher peak at the center of the
   // scanline than for a wider beam.
 #ifdef USEGAUSSIAN
-  vec4 wid = 0.3 + 0.1 * pow(color, vec4_splat(3.0));
+  vec4 wid = spot_size.x + spot_growth.x * pow(color, vec4_splat(spot_growth_power.x));
   vec4 weights = vec4(distance / wid);
-  return 0.4 * exp(-weights * weights) / wid;
+  float maxwid = spot_size.x + spot_growth.x;
+  float norm = maxwid / ( 1.0 + 2.0 * exp(-1.0/(maxwid*maxwid)) );
+  return norm * exp(-weights * weights) / wid;
 #else
   vec4 wid = 2.0 + 2.0 * pow(color, vec4_splat(4.0));
-  vec4 weights = vec4_splat(distance / 0.3);
+  vec4 weights = vec4_splat(abs(distance) / 0.3);
   return 1.4 * exp(-pow(weights * inversesqrt(0.5 * wid), wid)) / (0.6 + 0.2 * wid);
 #endif
 }
 
+vec4 cubic(vec4 x, float B, float C)
+{
+  // https://en.wikipedia.org/wiki/Mitchell%E2%80%93Netravali_filters
+  vec2 a = x.yz; // components in [0,1]
+  vec2 b = x.xw; // components in [1,2]
+  vec2 a2 = a*a;
+  vec2 b2 = b*b;
+  a = (2.0-1.5*B-1.0*C)*a*a2 + (-3.0+2.0*B+C)*a2 + (1.0-(1.0/3.0)*B);
+  b = ((-1.0/6.0)*B-C)*b*b2 + (B+5.0*C)*b2 + (-2.0*B-8.0*C)*b + ((4.0/3.0)*B+4.0*C);
+  return vec4(b.x,a.x,a.y,b.y);
+}
+
+vec4 x_coeffs(vec4 x, float pos_x)
+{
+  if (u_interp.x < 0.5) { // box
+    float wid = length(vec2(dFdx(pos_x),dFdy(pos_x)));
+    float dx = clamp((0.5 + 0.5*wid - x.y)/wid, 0.0, 1.0);
+    return vec4(0.0,dx,1.0-dx,0.0);
+  } else if (u_interp.x < 1.5) { // linear
+    return vec4(0.0, 1.0-x.y, 1.0-x.z, 0.0);
+  } else if (u_interp.x < 2.5) { // Lanczos
+    // Prevent division by zero.
+    vec4 coeffs = FIX(PI * x);
+    // Lanczos2 kernel.
+    coeffs = 2.0 * sin(coeffs) * sin(coeffs / 2.0) / (coeffs * coeffs);
+    // Normalize.
+    coeffs /= dot(coeffs, vec4_splat(1.0));
+    return coeffs;
+  } else if (u_interp.x < 3.5) { // Catmull-Rom
+    return cubic(x,0.0,0.5);
+  } else if (u_interp.x < 4.5) { // Mitchell-Netravali
+    return cubic(x,1.0/3.0,1.0/3.0);
+  } else if (u_interp.x < 5.5) { // B-spline
+    return cubic(x,1.0,0.0);
+  }
+}
```

Comment:
This function fails to compile because there’s a possibility of it falling off and not hitting a `return` statement if `u_interp.x` is greater than 5.5.  As a workaround, I’ve commented out the last `if` so it will default to B-spline in a860495f6c485ec69f92586254b9f698bfb3d553 (same applies to the equivalent function in fs_crt-geom.sc).